akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -433,14 +437,24 @@ object ShardCoordinator {
 
     def receive = {
       case BeginHandOffAck(`shard`) =>
-        remaining -= sender()
-        if (remaining.isEmpty) {
-          from ! HandOff(shard)
-          context.become(stoppingShard, discardOld = true)
-        }
+        log.debug("BeginHandOffAck for shard [{}] received from {}", shard, sender())
```

Comment:
```suggestion
        log.debug("'BeginHandOffAck' for shard [{}] received from {}.", shard, sender())
```

Reply:
This means receiving a `BeginHandOffAck` calls an `unwatch` in the `acked` function, which makes sense for receiving `Terminated` but less so here.